lib/utils.js

```diff
@@ -249,6 +249,8 @@ function merge(/* obj1, obj2, obj3, ... */) {
   function assignValue(val, key) {
     if (typeof result[key] === 'object' && typeof val === 'object') {
       result[key] = merge(result[key], val);
+    } else if (key === 'headers') {
```

Comment:
`assignValue` is a general-purpose utility function. I don't believe it should handle `headers` differently than any other keys.

Reply:
good point, i'll look else where to do a really tiny edit. IMO this is not something that should require a whole lotta code and/or another dependency..

Reply:
how do you feel about this more generalized conditional?

Reply:
added `Clone` function which gets called during `axios.create` to truly clone the `defaults` object. `utils.merge` changes have been undone.